packages/extension-base/src/background/handlers/State.ts

```diff
@@ -158,31 +158,29 @@ export default class State {
     ]);
   }
 
-  public async updateCurrentTabsUrl (urls: string[]) {
+  public updateActiveTabUrl (url: string | undefined) {
+    this.#activeTabUrl = url;
+  }
+
+  public async getConnectedActiveTabUrl () {
     const authUrls = await this.getAuthUrls();
 
-    const connectedTabs = urls.map((url) => {
-      let strippedUrl = '';
+    if (!this.#activeTabUrl) {
+      return undefined;
+    }
 
-      // the assert in stripUrl may throw for new tabs with "chrome://newtab/"
-      try {
-        strippedUrl = new URL(url).origin;
-      } catch (e) {
-        console.error(e);
-      }
+    try {
+      // may throw for new tabs with "chrome://newtab/"
+      const rawUrl = new URL(this.#activeTabUrl).origin;
 
-      // return the stripped url only if this website is known
-      return !!strippedUrl && authUrls[strippedUrl]
-        ? strippedUrl
-        : undefined;
-    })
-      .filter((value) => !!value) as string[];
+      const isConnected = authUrls[rawUrl];
 
-    this.#connectedTabsUrl = connectedTabs;
-  }
+      return isConnected ? rawUrl : undefined;
+    } catch (e) {
+      console.error('Error calculating connected active tab url:', e);
 
-  public getConnectedTabsUrl () {
-    return this.#connectedTabsUrl;
+      return undefined;
+    }
   }
```

Comment:
This part is where the actual fix lies: Previously, the _current active tab_ was registered upon active tab changes (all the `chrome.xxx` listeners that `updateCurrentTabsUrl()` was registered as), but not upon the auth url changes, meaning that every time the auth urls changed, `getConnectedTabsUrl()` would start returning possibly incorrect results. The fix is based on switching to calculating the _current active tab_ laziliy (moved from the `updateCurrentTabsUrl()` _setter_ to the `getConnectedActiveTabUrl()` _getter_) so that the calculations are always performed on the freshest data.